CMakeLists.txt

```diff
@@ -175,6 +175,7 @@ list(APPEND OT_PUBLIC_INCLUDES ${PROJECT_SOURCE_DIR}/include)
 
 if(OT_PLATFORM STREQUAL "posix")
     target_include_directories(ot-config INTERFACE ${PROJECT_SOURCE_DIR}/src/posix/platform)
+    target_compile_definitions(ot-config INTERFACE OPENTHREAD_PLATFORM_POSIX=1)
```

Comment:
I think this is missing in CMake since it is defined in https://github.com/openthread/openthread/blob/b9a318092c06244ac8254ca32c31f2f46e01edd7/configure.ac#L882